GetDKAN/dkan
# Update getExtension() and createResources() functions

Connects https://github.com/NuCivic/cadepttech/issues/111

Harvesting issues:
- A bad resource will block the rest of the resources from being attached to the dataset.
- The getExtension() function is coming up with false extensions

## How to reproduce

1.  Create a harvest source from https://api.myjson.com/bins/kc1dz
This source has one dataset with three resource files.
(if that link no longer works create a new source on http://myjson.com with this file:
[test-json.txt](https://github.com/GetDKAN/dkan/files/1952843/test-json.txt) )

2. Harvest.
 -- You will see this error: 
```
Resource remote url (https://data.cnra.ca.gov/dataset/6fc2f546-2a84-401f-ac5e-59bbd42da793/resource/3e176f27-6f1d-4b91-8e6f-072bdde0c28b/download/stations.csv) extension (xsl) not allowed
```
--- The dataset harvested will have 0 resources. 
--- The first resource (TEST - Workforce By Generation (2011-2015)) is created but will not be connected to the dataset.
--- The second resource will not be created because of the inaccurate extension assigned (xsl)
--- The third resource will not be created because the download link is bad.

## QA Steps
1. Create the harvest source from above
2. Harvest.
3. Confirm that two of the resources are created and linked to the dataset with the correct extensions. 
4. Check the harvest error log `node/{nid}/harvest-errors` and confirm you see a the 'Cannot create the resource. Invalid data value given.' error because of the bad resource url.
